File: app-android/app/src/main/java/com/health2u/presentation/editprofile/EditProfileState.kt
package com.health2u.presentation.editprofile

data class EditProfileState(
    val name: String = "",
    val email: String = "",
    val phone: String = "",
    val dateOfBirth: String = "",
    val profilePictureUrl: String? = null,
    val isLoading: Boolean = true,
    val isSaving: Boolean = false,
    val nameError: String? = null,
    val emailError: String? = null,
    val saveSuccess: Boolean = false,
    val error: String? = null
)

File: app-ios/Sources/Health2u/Presentation/EditProfile/EditProfileState.swift
import Foundation

public struct EditProfileState: Equatable {
    public var name: String = ""
    public var email: String = ""
    public var phone: String = ""
    public var dateOfBirth: Date? = nil
    public var gender: String = ""
    public var heightCm: String = ""
    public var weightKg: String = ""
    public var bloodType: String = ""
    public var userId: String = ""
    public var isLoading: Bool = false
    public var isSaving: Bool = false
    public var error: String? = nil
    public var didSave: Bool = false
    public var isUploadingPhoto: Bool = false

    public init() {}
}
